golang/cosmos/x/vtransfer/keeper/keeper.go

```diff
@@ -0,0 +1,197 @@
+package keeper
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	capabilitytypes "github.com/cosmos/cosmos-sdk/x/capability/types"
+
+	"github.com/Agoric/agoric-sdk/golang/cosmos/x/vibc"
+	transfertypes "github.com/cosmos/ibc-go/v6/modules/apps/transfer/types"
+	channeltypes "github.com/cosmos/ibc-go/v6/modules/core/04-channel/types"
+	porttypes "github.com/cosmos/ibc-go/v6/modules/core/05-port/types"
+	host "github.com/cosmos/ibc-go/v6/modules/core/24-host"
+	ibcexported "github.com/cosmos/ibc-go/v6/modules/core/exported"
+)
+
+var (
+	_ porttypes.ICS4Wrapper = (*Keeper)(nil)
+)
+
+const (
+	targetedStoreKeyPrefix = "tgt/"
+	targetedSentinel       = "\x01"
+)
+
+// Keeper maintains the link to data storage and exposes getter/setter methods for the various parts of the state machine
+type Keeper struct {
+	vibc.Keeper
```

Comment:
I don't like having a `Keeper` field on a `Keeper` type, especially when it's actually a very different type.
```suggestion
	vibcKeeper vibc.Keeper
```